examples/consumer_at_least_once_example/consumer.go

```diff
@@ -121,15 +106,20 @@ func main() {
 					fmt.Printf("%% Headers: %v\n", e.Headers)
 				}
 
-				// We can do any async operation here, because the
-				// offset will not be commited until the next loop
-				// we can spend all the time will need in this step.
+				// We can do any long running async operation here, because the next
+				// offset will not be available for commit until we call `c.StoreOffsets(...)`
 				_ = e
-				// Then after a successful transaction we store the
-				// last message offset inside a variable to
-				// be committed in the next loop.
-				lastMessage = e
-				msgsSinceLastCommit++
+				// In this example we store the next offset that will be committed
+				// to the offset store according to `auto.commit.interval.ms` or manual
+				// offset-less Commit().
+				// In this example we choose to terminate
+				// the application if the offset can't be stored.
+				offsets := []kafka.TopicPartition{e.TopicPartition}
+				offsets[0].Offset++
```

Comment:
Also why are we incrementing the offset, suppose the offset delivered was `15`, shouldn't we be committing offset `15` instead of `16`. Correct me if I am wrong

Reply:
Hi there, this is a little confusing, but offset means next offset instead of last one. https://github.com/confluentinc/confluent-kafka-go/issues/350

Reply:
ah - so I wasn't mad - you do commit the next offset

Reply:
Yes @alexec. I understand that this solution is a bit confusing but I couldn't find a better solution.

Reply:
There's a new StoreMessage() method that you can use instead, which abstracts these details.